File: runtime/ruby/units/wasi/fd_tell.rb
# requires: memory/i64_store, rt/m64
def wasi_fd_tell(fd, out_ptr)
  io = @fds[fd]
  return ERRNO_BADF if io.nil? || io.is_a?(WasiDir)
  return ERRNO_SPIPE if @std_ios.include?(io)
  @memory.i64_store(out_ptr, Rt.m64(io.tell))
  ERRNO_SUCCESS
rescue SystemCallError
  ERRNO_IO
end

File: crates/dewasm-backend-python/units/wasi/fd_tell.py
# requires: memory/ids
def wasi_fd_tell(self, fd, out_ptr):
    io = self.fds.get(fd)
    if io is None or isinstance(io, self.WasiDir):
        return self.ERRNO_BADF
    if io in self.std_ios:
        return self.ERRNO_SPIPE
    try:
        self.memory.ids(out_ptr, io.tell() & Rt.M64)
    except OSError:
        return self.ERRNO_IO
    return self.ERRNO_SUCCESS
